x-pack/plugins/apm/public/components/shared/transactions_table/get_columns.tsx

```diff
@@ -49,19 +54,68 @@ export function getColumns({
   transactionGroupDetailedStatistics,
   comparisonEnabled,
   shouldShowSparkPlots = true,
+  showAlertsColumn,
   offset,
   transactionOverflowCount,
+  link,
+  query,
 }: {
   serviceName: string;
   latencyAggregationType?: LatencyAggregationType;
   transactionGroupDetailedStatisticsLoading: boolean;
   transactionGroupDetailedStatistics?: TransactionGroupDetailedStatistics;
   comparisonEnabled?: boolean;
   shouldShowSparkPlots?: boolean;
+  showAlertsColumn: boolean;
   offset?: string;
   transactionOverflowCount: number;
+  link: any;
+  query: TypeOf<ApmRoutes, '/services/{serviceName}/overview'>['query'];
 }): Array<EuiBasicTableColumn<ServiceTransactionGroupItem>> {
   return [
+    ...(showAlertsColumn
+      ? [
+          {
+            field: 'alertsCount',
+            sortable: true,
+            name: i18n.translate(
+              'xpack.apm.transactionsTableColumnName.alertsColumnLabel',
+              { defaultMessage: 'Active alerts' }
+            ),
+            width: `${unit * 6}px`,
+            render: (_, { alertsCount }) => {
+              if (!alertsCount) {
+                return null;
+              }
+              return (
+                <EuiToolTip
+                  position="bottom"
+                  content={i18n.translate(
+                    'xpack.apm.home.transactionsTableColumnName.tooltip.activeAlertsExplanation',
+                    {
+                      defaultMessage: 'Active alerts',
+                    }
+                  )}
+                >
+                  <EuiBadge
+                    iconType="warning"
+                    color="danger"
+                    href={link('/services/{serviceName}/alerts', {
+                      path: { serviceName },
+                      query: {
+                        ...query,
+                        alertStatus: ALERT_STATUS_ACTIVE,
```

Comment:
I'm thinking that the experience would be better if we also filter transaction name and type when clicking on the active alerts on the transaction table.

Reply:
Agreed! The alerts table should only show alerts for transactions with the same name and type.